web/web/models/notification.py

```diff
@@ -47,6 +48,15 @@ class Notification(Model):
                                        name='_alert_type_id_email_uc'), )
 
     # Methods
+    @validates('email')
+    def empty_string_to_null(self, key, value):
+        # converts empty string to null as field is not nullable
+        if isinstance(value,str) and value == '':
+            return None
+        else:
+            return value
```

Comment:
Couple things here that don't make sense to me:
- If the field is not nullable, why are we returning `None`? Is the intention to actually let the DB throw the exception? If so, may be good to comment that
- Why the instance check? Coercion should cover all instance cases
- This check fails if I send a string with multiple whitespaces, like: `'       '` vs `''`, so it may be worth trimming.

Reply:
fixed